### ISSUE TITLE
Working with global variabels, allure reporter

### ISSUE THREAD

**Issue by qatechniker** [yandex/mocha-parallel-tests#162]
Hello,
I am using  mocha-multi-reporters to generate couple reports. 
Allure one in list - mocha-allure-reporter (inside creates global variable - allure).
I am using global allure to add custom steps to reporter:
`declare let allure: any;`
and then:
`allure._allure.startStep(title);`
It works with mocha, but I got error once start with mocha-parallel-tests:
ReferenceError: allure is not defined

Finally I got my allure report, but without custom steps.

Can you comment, please?

**Comment by 1999** [comment]
@qatechniker Interesting. Allure reporter is said to be supported since `mocha-parallel-tests` version 2.0.1: here's the test which demoes this: https://github.com/yandex/mocha-parallel-tests/blob/master/test/reporter-custom-allure/index.sh#L8
Which `mocha-parallel-tests` and `mocha-allure-reporter` version are you using?

**Comment by qatechniker** [comment]
Hello @1999 , reports generating finally from describe and it, but custom steps ...  no

"mocha-allure-reporter": "^1.4.0",
"mocha": "^5.1.1",
"mocha-parallel-tests": "^2.0.2",
"mocha-multi-reporters": "^1.1.7",

**Comment by 1999** [comment]
@qatechniker Unfortunately I don't have much experience with Allure. Can you show what's custom step and how you launch test when you run it with mocha?

**Comment by qatechniker** [comment]
Sure.
1) It is ts project with OOP approach
2) My tests start command line:
    "rest-api-tests-parallel": "./node_modules/.bin/mocha-parallel-tests --max-parallel 8 --reporter mocha-multi-reporters --reporter-options configFile=./pathToFile/reporter-config.json --timeout 1000000 --recursive ./dist-rest-api-test/rest-api-tests/src/"
My mocha-multi-reporters config:
{
    "reporterEnabled": "good-mocha-html-reporter, mocha-junit-reporter, mocha-allure-reporter",
    "mochaJunitReporterReporterOptions": {
        "mochaFile": "junit-report.xml"
    }
}
This is enough to get allure report based on desc and it of mocha (desc -> suit, it -> test).
3) To add custom steps, I use code:
```
declare let allure: any;

/**
 * Class to wrap and collect html reporter functions
 */

export class HtmlReporter {

    static startStep(title: string) {
        try {
            allure._allure.startStep(title);
        } catch (e) {
            // console.log(e);
            return;
        }
    }

    static endStep() {
        try {
            allure._allure.endStep();
        } catch (e) {
            return;
        }
    }
```
Call methods:
HtmlReporter.startStep('Test title');
// code here
HtmlReporter.endStep();


This stuff works with mocha, I can see nice reports with my custom steps.
I can not see steps if run with mocha parallel.

**Comment by qatechniker** [comment]
Em, did not get the question....
Yes - I am starting tests with mocha-parallel-tests as you can see in my start tests string

**Comment by qatechniker** [comment]
mocha-allure-reporter during start creates global variable allure, which I am using then in custom steps

**Comment by 1999** [comment]
Okay, I investigated this case a bit and came up with a limitations page: https://github.com/yandex/mocha-parallel-tests/wiki/Limitations
I also added a link to it to the project's README.

TL;DR: no good news here: custom variables are pretty hard to support in a multi-process mocha tests execution environment.
